### PR TITLE
Added browser tests

### PR DESCRIPTION
**Pull request by nginnever**

### CODE REVIEW

**Comment by daviddias** [comment]
@nginnever you need to update the travis.yml file to support launching firefox https://github.com/vijayee/js-ipfs-merkle-dag/blob/master/.travis.yml

File: .travis.yml [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
+branches:
+  only:
+    - master
+
+before_install:
+  - npm i -g npm
+  # Workaround for a permissions issue with Travis virtual machine images
+script:
+  - npm test
+
```
Comment (daviddias) [inline .travis.yml:11]: try adding 

```
addons:
  firefox: 'latest'
```

**Comment by daviddias** [comment]
Also, why did you had to copy all the test data that we use for data-importing? Couldn't use the same test repo that was already present here?

**Comment by daviddias** [comment]
We've greeeeeen!

So, what is missing is:
- [x] remove all unnecessary test data that was added
- [x] pass the ipfs-repo to the tests, avoiding to have test duplicates

File: package.json [hunk 2]
```diff
@@ -16,28 +18,43 @@
...
-    "type" : "git"
-  , "url" : "https://github.com/vijayee/js-ipfs-merkle-dag.git"
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/vijayee/js-ipfs-merkle-dag.git"
   },
   "dependencies": {
     "bl": "^1.0.0",
     "ipfs-blocks": "^0.1.0",
+    "ipfs-unixfs": "^0.1.0",
```
Comment (daviddias) [inline package.json]: do we need this?

File: package.json [hunk 3]
```diff
@@ -16,28 +18,43 @@
...
+    "karma-sourcemap-loader": "^0.3.7",
+    "karma-spec-reporter": "0.0.24",
+    "karma-webpack": "^1.7.0",
+    "lodash": "^4.6.1",
     "mocha": "^2.4.5",
     "ncp": "^2.0.0",
     "pre-commit": "^1.1.2",
     "rimraf": "^2.5.0",
-    "standard": "^5.4.1"
+    "standard": "^5.4.1",
```
Comment (daviddias) [inline package.json]: can we up this to standard 6?

File: package.json [hunk 4]
```diff
@@ -16,28 +18,43 @@
...
+    "transform-loader": "^0.2.3",
+    "webpack": "github:diasdavid/webpack#webpack-1"
```
Comment (daviddias) [inline package.json]: following data-importing, let's do webpack 2 always from now on

File: src/dag-node.js [hunk 5]
```diff
@@ -53,7 +53,7 @@ function DAGNode (data, links) {
   this.addRawLink = (link) => {
     encoded = null
     this.links.push(new DAGLink(link.name, link.size, link.hash))
-    stable.inplace(this.links,linkSort)
+    stable.inplace(this.links)
```
Comment (daviddias) [inline src/dag-node.js]: why no more link sorting?

File: package.json (hunk 2 above)
Comment (nginnever) [inline package.json]: not that i can see

File: src/dag-node.js [hunk 6]
```diff
@@ -151,7 +151,7 @@ function DAGNode (data, links) {
 
   // unMarshal - decodes a protobuf into a DAGNode
   // TODO: this would make more sense as an utility
-  this.unMarshal = (data) => {
+  this.unMarshal = function (data) {
```
Comment (daviddias) [inline src/dag-node.js:154]: because of the "Buffer is undefined issue" ?

File: tests/merkle-dag-tests.js [hunk 7]
```diff
@@ -0,0 +1,262 @@
+/* globals describe, it */
```
Comment (daviddias) [inline tests/merkle-dag-tests.js]: `/* eslint-env mocha */`

File: tests/index.js [hunk 8]
```diff
@@ -1,13 +1,13 @@
 /* globals describe, before, after */
```
Comment (daviddias) [inline tests/index.js]: `/* eslint-env mocha */`

**Comment by daviddias** [comment]
mind squashing when it gets ready for the merge?

File: src/dag-node.js (hunk 5 above)
Comment (nginnever) [inline src/dag-node.js]: was flagged in standard for no space so i took it off and yielded that same results without a comparator function. Not sure why so putting it back.

Comment (nginnever) [inline src/dag-node.js]: probably since our link sort exhibits the default behaviour of stable

**Pull request merged**